x-pack/plugins/maps/public/components/layer_addpanel/index.js

```diff
@@ -43,14 +40,21 @@ function mapDispatchToProps(dispatch) {
       dispatch(setSelectedLayer(layer.getId()));
       dispatch(setTransientLayer(layer.getId()));
     },
-    removeTransientLayer: () => {
+    addImportedLayer: async layer => {
```

Comment:
It was exactly how you're describing at one point before it became apparent the 2 workflows needed to diverge. Since we've removed some of the logic specific to importing, it might be they need to converge again. I'll take a look

Reply:
I ended up refactoring this and consolidating the logic